test_upload.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+from unittest.mock import patch
+
+
+@pytest.fixture
+def mock_env(monkeypatch):
+    monkeypatch.setenv("GITHUB_WORKSPACE", "/fake/workspace")
+    monkeypatch.setenv("COMPONENTS_NAMESPACE", "test_ns")
+    monkeypatch.setenv("COMPONENTS_DIRECTORIES", "comp1;comp2")
```

Comment:
shall we test the newly added use case `new_name:folder;`?

Reply:
@hfudev Should be done now